unity-renderer/Assets/DCLPlugins/ECS7/Systems/PlayerSystem/ECSPlayerTransformSystem.cs

```diff
@@ -1,78 +1,89 @@
 
-        public static Action CreateSystem(IECSComponentWriter componentsWriter)
+        public void Dispose()
         {
-            var state = new State()
-            {
-                avatarTransform = DataStore.i.world.avatarTransform,
-                rendererState = CommonScriptableObjects.rendererState,
-                worldOffset = CommonScriptableObjects.worldOffset,
-                loadedScenes = DataStore.i.ecs7.scenes,
-                componentsWriter = componentsWriter
-            };
-            return () => Update(state);
+            loadedScenes.OnAdded -= LoadedScenesOnOnAdded;
         }
 
-        private static void Update(State state)
+        public void Update()
         {
-            if (!state.rendererState.Get())
+            if (!rendererState.Get())
                 return;
 
-            Transform avatarT = state.avatarTransform.Get();
+            Transform avatarT = avatarTransform.Get();
 
-            UnityEngine.Vector3 avatarPosition = avatarT.position;
+            Vector3 avatarPosition = avatarT.position;
             Quaternion avatarRotation = avatarT.rotation;
 
-            if (state.lastAvatarPosition == avatarPosition && state.lastAvatarRotation == avatarRotation)
+            if (!newSceneAdded && lastAvatarPosition == avatarPosition && lastAvatarRotation == avatarRotation)
             {
                 return;
             }
 
-            state.lastAvatarPosition = avatarPosition;
-            state.lastAvatarRotation = avatarRotation;
-            UnityEngine.Vector3 worldOffset = state.worldOffset.Get();
+            newSceneAdded = false;
+
+            lastAvatarPosition = avatarPosition;
+            lastAvatarRotation = avatarRotation;
+            Vector3 worldOffset = this.worldOffset.Get();
 
-            var loadedScenes = state.loadedScenes;
-            var componentsWriter = state.componentsWriter;
+            IReadOnlyList<IParcelScene> loadedScenes = this.loadedScenes;
+            var componentsWriter = this.componentsWriter;
 
             IParcelScene scene;
+
             for (int i = 0; i < loadedScenes.Count; i++)
             {
                 scene = loadedScenes[i];
 
                 var transform = TransformHelper.SetTransform(scene, ref avatarPosition, ref avatarRotation, ref worldOffset);
 
                 componentsWriter.PutComponent(scene.sceneData.sceneNumber, SpecialEntityId.PLAYER_ENTITY, ComponentID.TRANSFORM,
-                    transform, state.timeStamp, ECSComponentWriteType.SEND_TO_SCENE);
+                    transform, timeStamp, ECSComponentWriteType.SEND_TO_SCENE);
 
                 componentsWriter.PutComponent(scene.sceneData.sceneNumber, SpecialEntityId.INTERNAL_PLAYER_ENTITY_REPRESENTATION, ComponentID.TRANSFORM,
                     transform, ECSComponentWriteType.EXECUTE_LOCALLY);
             }
-            state.timeStamp++;
+
+            timeStamp++;
+        }
+
+        private void LoadedScenesOnOnAdded(IParcelScene obj)
```

Comment:
Maybe one `On` in the name is enough?

Reply:
Yes I think so too, renamed it [here](https://github.com/decentraland/unity-renderer/pull/3700/commits/1ec6ed8b26e288c53393a08c8729a59f76738f47)